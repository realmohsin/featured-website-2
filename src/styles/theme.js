import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
