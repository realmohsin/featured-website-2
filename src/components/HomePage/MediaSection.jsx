import React from 'react'
import clsx from 'clsx'
import { makeStyles, useTheme } from '@material-ui/core/styles'

import { Grid, Container } from '@material-ui/core'
import Button from '../Button'

const useStyles = makeStyles(theme => ({
  blogMediaSection: {
    padding: '10rem 0 12rem'
  },
  titleList: {
    display: 'flex',
    justifyContent: 'center'
  },
  title: {
    ...theme.custom.title2,
    margin: '0 3rem'
  },
  activeTitle: {
    color: theme.palette.primary.main,
    textDecoration: 'underline'
  },

  mediaSection: {},

  itemDate: {
    color: theme.palette.primary.main,
    marginBottom: '1.2rem'
  },
  button: {
    margin: '3rem auto'
  },
  mediaItem: {
    padding: '3rem 4rem 3rem'
  },
  videoContainer: {
    position: 'relative',
    width: '100%',
    paddingTop: '56.25%', // for youtube embeds,
    marginBottom: '1.6rem'
  },
  iframeYT: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    width: '100%',
    height: '100%'
  },
  mediaItemTitle: {
    ...theme.custom.title5,
    color: theme.palette.secondary.dark,
    letterSpacing: '0.2px',
    marginTop: '-0.75rem'
  }
}))

const BlogMediaSection = props => {
  const theme = useTheme()
  const classes = useStyles()

  return (
    <section className={classes.blogMediaSection}>
      <Container>
        <h2 className={classes.title}>In the Media</h2>
        <div className={classes.mediaSection}>
          <Grid container>
            <Grid
              item
              xs={12}
              md={6}
              className={clsx(classes.mediaItem, classes.leftMediaItem)}
            >
              <div className={classes.videoContainer}>
                <iframe
                  title=''
                  src='https://www.youtube.com/embed/Bva8ARJrCog'
                  frameborder='0'
                  allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
                  allowfullscreen
                  className={classes.iframeYT}
                />
              </div>
              <p className={classes.itemDate}>
                <time>11.07.19</time>
              </p>
              <h5 className={classes.mediaItemTitle}>
                DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES
                OUT OF CHINA – 8/23/2019
              </h5>
            </Grid>
            <Grid
              item
              xs={12}
              md={6}
              className={clsx(classes.mediaItem, classes.rightMediaItem)}
            >
              <div className={classes.videoContainer}>
                <iframe
                  title=''
                  src='https://www.youtube.com/embed/Bva8ARJrCog'
                  frameborder='0'
                  allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
                  allowfullscreen
                  className={classes.iframeYT}
                />
              </div>
              <p className={classes.itemDate}>
                <time>11.07.19</time>
              </p>
              <h5 className={classes.mediaItemTitle}>
                DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES
                OUT OF CHINA – 8/23/2019
              </h5>
            </Grid>
            <Grid
              item
              xs={12}
              md={6}
              className={clsx(classes.mediaItem, classes.leftMediaItem)}
            >
              <div className={classes.videoContainer}>
                <iframe
                  title=''
                  src='https://www.youtube.com/embed/Bva8ARJrCog'
                  frameborder='0'
                  allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
                  allowfullscreen
                  className={classes.iframeYT}
                />
              </div>
              <p className={classes.itemDate}>
                <time>11.07.19</time>
              </p>
              <h5 className={classes.mediaItemTitle}>
                DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES
                OUT OF CHINA – 8/23/2019
              </h5>
            </Grid>
            <Grid
              item
              xs={12}
              md={6}
              className={clsx(classes.mediaItem, classes.rightMediaItem)}
            >
              <div className={classes.videoContainer}>
                <iframe
                  title=''
                  src='https://www.youtube.com/embed/Bva8ARJrCog'
                  frameborder='0'
                  allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
                  allowfullscreen
                  className={classes.iframeYT}
                />
              </div>
              <p className={classes.itemDate}>
                <time>11.07.19</time>
              </p>
              <h5 className={classes.mediaItemTitle}>
                DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES
                OUT OF CHINA – 8/23/2019
              </h5>
            </Grid>
          </Grid>
          <Button
            isLink
            text='Visit our Media Page'
            extraClassName={classes.button}
          />
        </div>
      </Container>
    </section>
  )
}

export default BlogMediaSection
