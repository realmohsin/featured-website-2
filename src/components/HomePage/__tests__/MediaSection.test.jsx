import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import MediaSection from "../MediaSection";

// Mock gatsby-plugin-image
jest.mock("gatsby-plugin-image", () => ({
  StaticImage: ({ src, alt, className, ...props }) => (
    <img src={src} alt={alt} className={className} {...props} />
  ),
  getImage: jest.fn((image) => image)
}));

// Mock the Button component
jest.mock("../../Button", () => ({
  __esModule: true,
  default: ({ text, href, ...props }) => (
    <a href={href} {...props}>
      {text}
    </a>
  ),
}));

const theme = createMuiTheme({
  palette: {
    primary: { main: "#0E5C60", light: "#4A8B8F" },
    secondary: { main: "#CCAB63" },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
  custom: {
    title2: {
      fontSize: "5.6rem",
      fontWeight: "bold",
    },
    title3: {
      fontSize: "3.2rem",
    },
    title5: {
      fontSize: "1.8rem",
    },
  },
});

// Mock window.matchMedia which is not available in JSDOM
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("MediaSection Component", () => {
  it("renders media section with all content", () => {
    const { container } = renderWithTheme(<MediaSection />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
