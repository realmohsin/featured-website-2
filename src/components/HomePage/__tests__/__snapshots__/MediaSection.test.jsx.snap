// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaSection Component renders media section with all content 1`] = `
<section
  class="makeStyles-blogMediaSection-1"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="makeStyles-title-3"
    >
      In the Media
    </h2>
    <div
      class="makeStyles-mediaSection-5"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root makeStyles-mediaItem-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-videoContainer-9"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              class="makeStyles-iframeYT-10"
              frameborder="0"
              src="https://www.youtube.com/embed/Bva8ARJrCog"
              title="Domenic Romano on Fox Business News - Trump Orders US Companies Out of China - 8/23/2019"
            />
          </div>
          <p
            class="makeStyles-itemDate-6"
          >
            <time>
              11.07.19
            </time>
          </p>
          <h5
            class="makeStyles-mediaItemTitle-11"
          >
            DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES OUT OF CHINA – 8/23/2019
          </h5>
        </div>
        <div
          class="MuiGrid-root makeStyles-mediaItem-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-videoContainer-9"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              class="makeStyles-iframeYT-10"
              frameborder="0"
              src="https://www.youtube.com/embed/oiUsGImX4uY"
              title="Domenic Romano on Fox Business News - Second Interview"
            />
          </div>
          <p
            class="makeStyles-itemDate-6"
          >
            <time>
              11.07.19
            </time>
          </p>
          <h5
            class="makeStyles-mediaItemTitle-11"
          >
            DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES OUT OF CHINA – 8/23/2019
          </h5>
        </div>
        <div
          class="MuiGrid-root makeStyles-mediaItem-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-videoContainer-9"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              class="makeStyles-iframeYT-10"
              frameborder="0"
              src="https://www.youtube.com/embed/NJELMaBtUFY"
              title="Domenic Romano on Fox Business News - Third Interview"
            />
          </div>
          <p
            class="makeStyles-itemDate-6"
          >
            <time>
              11.07.19
            </time>
          </p>
          <h5
            class="makeStyles-mediaItemTitle-11"
          >
            DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES OUT OF CHINA – 8/23/2019
          </h5>
        </div>
        <div
          class="MuiGrid-root makeStyles-mediaItem-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <div
            class="makeStyles-videoContainer-9"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              class="makeStyles-iframeYT-10"
              frameborder="0"
              src="https://www.youtube.com/embed/W4HDYa9kk_g"
              title="Domenic Romano on Fox Business News - Fourth Interview"
            />
          </div>
          <p
            class="makeStyles-itemDate-6"
          >
            <time>
              11.07.19
            </time>
          </p>
          <h5
            class="makeStyles-mediaItemTitle-11"
          >
            DOMENIC ROMANO ON FOX BUSINESS NEWS – TRUMP ORDERS US COMPANIES OUT OF CHINA – 8/23/2019
          </h5>
        </div>
      </div>
    </div>
  </div>
</section>
`;
